### ISSUE TITLE
[Request]: Battle.net

### ISSUE THREAD

**Issue by EstebanMunoz** [Calinou/scoop-games#1016]
### Prerequisites

- [X] I have searched all issues/PRs to ensure it has not already been reported or fixed.

### Criteria

- [X] Reasonably well-known and widely used
- [X] A game or a game-related tool
- [X] English interface (or at least English documentation)
- [X] Latest stable or development version
- [X] Full version (i.e. not a trial version)
- [ ] Fairly standard install (e.g. uses a version-specific download URL, no elaborate pre/post install scripts)

### Name

battlenet

### Description

Blizzard's official games client.

### Homepage

https://battle.net/

### Download Link(s)

https://eu.battle.net/download/getInstaller?os=win&installer=Battle.net-Setup.exe

### Some Indication of Popularity/Repute

wikipedia.org/wiki/Battle.net

**Comment by EstebanMunoz** [comment]
I've been serching about this, and there are 2 major concerns regarding the addition of battle.net. The first one is the lack of a stable hash for comparison, since every new download gives an installer with a different hash. The other problem is the lack of silent install switches in the installer, however I think it is possible to install and uninstall the progam without user intervention, and I've seen other programs included in this bucket allowed to show windows during the installation process so this shouldn't be a problem.

In case it is possible to include an app without a hash given the file is downloaded directly from the provider, there is a chance to include this app in the bucket. If that's not the case, it is currently impossible to do so.

**Comment by Calinou** [comment]
> In case it is possible to include an app without a hash given the file is downloaded directly from the provider, there is a chance to include this app in the bucket. If that's not the case, it is currently impossible to do so.

You can specify the version identifier as `nightly` to skip hash checking (and omit any hashes in the manifest), but this may cause issues if the app needs to be updated and doesn't update correctly if written to a custom location.

**Comment by EstebanMunoz** [comment]
Ok, so I tried to write a manifest, but in the process I encountered something a little weird. First I wrote a PowerShell script to install the app in a desired location, and everything went very well. Here it is:

```pwsh
Start-Process ./setup.exe --installpath="C:\Program Files (x86)\Battle.net"
$StartDate = Get-Date
$SetupProcess = Get-Process -name "Battle.net-Setup" -ErrorAction SilentlyContinue
DO {
    $SetupProcess = Get-Process -name "Battle.net-Setup" -ErrorAction SilentlyContinue
} Until ($null -ne $SetupProcess -or $StartDate.AddSeconds(10) -lt (Get-Date))

Wait-Process -id $SetupProcess.Id
$BattlenetProcess = Get-Process -Name "Battle.net" -ErrorAction SilentlyContinue
if ($null -eq $BattlenetProcess) {
    "Error"
} else {
    "Success"
    Stop-Process $BattlenetProcess -Force -ErrorAction SilentlyContinue
}
```

This is the explanation: it just starts the installer with a given path, and then tries to get the `Battle.net-Setup` process. When everything goes well, this process terminates after some `Battle.net` processes are started, so it is safe to search for the latter when the first has been terminated. Such behavior is important since there is no easy way to hide the installer window, and the user could potentially cancel the installation. When this happens, the `Battle.net` process is not started and an error is raised. This is also useful in the cases in which the installer doesn't start the installation and instead displays some "Try again" window, resulting in the `Battle.net` process not initializing.

So far so good. The problem here comes when I try to put this code in a manifest. Doing `sudo scoop install battlenet` does everything as expected, except for the fact that the `Battle.net-Setup` process is never encountered, and thus the rest of the code goes south. I could just remove the code related to that process, but in that scenario I wouldn't be capable to notice when the installation has been cancelled or terminated, or how much time to assign to the Timeout while looking for the `Battle.net` process, given it depends on the user machine and/or internet.

I don't know why this changes using scoop, so at this point I'm open to other ideas.
